webapp/src/components/Modals/BuyWithCryptoModal/ChainAndTokenSelector/ChainAndTokenSelector.tsx

```diff
@@ -79,8 +79,16 @@ const ChainAndTokenSelector = (props: Props) => {
     )
     // this sortes the tokens by USD balance
     filtered?.sort((a, b) => {
-      const aQuote = balances[a.address.toLowerCase()]?.quote ?? '0'
-      const bQuote = balances[b.address.toLowerCase()]?.quote ?? '0'
+      // MATIC has a different address in the balances array than in the tokens array
+      let tokenBalanceAddressA = a.address.toLocaleLowerCase()
```

Comment:
As we know the address in an hex code, what if we use:
```suggestion
      let tokenBalanceAddressA = a.address.toLowerCase()
```